pkg/threescale/connectors/backend/cached_backend.go

```diff
@@ -28,8 +28,12 @@ type Limit struct {
 	periodEnds int64
 }
 
-// CacheValue which should be stored in cache implementation
-type CacheValue struct {
+// CacheValue which should be stored in cache implementation maps an
+// application identifier to an Application which houses information  required for caching purpose
+type CacheValue map[string]*Application
+
+// Application defined under a 3scale service
+type Application struct {
```

Comment:
As it stands now we will be storing a pointer to the client in each application (within `ReportWithValues`). I think we'd rather have that pointer at a higher level, definitely no lower than service, perhaps at a higher level if we cache also by 3scale backend instance.